### ISSUE TITLE
Add Parallax Background effect for section

### ISSUE THREAD

**Issue by arielk** [elementor/elementor#72]

**Comment by 12steprecovery** [comment]
I investigated how the Sydney theme https://en-gb.wordpress.org/themes/sydney/ got its parallax effect (see http://demo.athemes.com/sydney/) to see if it could be added to Elementor.

I discovered that Sydney is using jQuery-Parallax https://github.com/IanLunn/jQuery-Parallax.

So I borrowed some code from Sydney and added it to the functions.php of my child theme as follows:

```
function jquery_parallax() {

   ?>
    <script language="JavaScript" type="text/javascript">
		

;(function($) {

   'use strict'

    var testMobile;
    var isMobile = {
        Android: function() {
            return navigator.userAgent.match(/Android/i);
        },
        BlackBerry: function() {
            return navigator.userAgent.match(/BlackBerry/i);
        },
        iOS: function() {
            return navigator.userAgent.match(/iPhone|iPad|iPod/i);
        },
        Opera: function() {
            return navigator.userAgent.match(/Opera Mini/i);
        },
        Windows: function() {
            return navigator.userAgent.match(/IEMobile/i);
        },
        any: function() {
            return (isMobile.Android() || isMobile.BlackBerry() || isMobile.iOS() || isMobile.Opera() || isMobile.Windows());
        }
    };

	var parallax = function() {
		testMobile = isMobile.any();
		if (testMobile == null) {
			$(".parallax").parallax("50%", 0.3);
		}
	};



	// Dom Ready
	$(function() {
		parallax();
   	});
})(jQuery);

/*
jQuery Parallax 1.1.3
Author: Ian Lunn
Plugin URL: http://www.ianlunn.co.uk/plugins/jquery-parallax/

Dual licensed under the MIT and GPL licenses:
http://www.opensource.org/licenses/mit-license.php
http://www.gnu.org/licenses/gpl.html
*/
!function(n){var t=n(window),e=t.height();t.resize(function(){e=t.height()}),n.fn.parallax=function(o,r,i){function u(){var i=t.scrollTop();l.each(function(t,u){var l=n(u),f=l.offset().top,s=a(l);i>f+s||f>i+e||l.css("backgroundPosition",o+" "+Math.round((l.data("firstTop")-i)*r)+"px")})}var a,l=n(this);l.each(function(t,e){$element=n(e),$element.data("firstTop",$element.offset().top)}),a=i?function(n){return n.outerHeight(!0)}:function(n){return n.height()},(arguments.length<1||null===o)&&(o="50%"),(arguments.length<2||null===r)&&(r=.1),(arguments.length<3||null===i)&&(i=!0),t.bind("scroll",u).resize(u),u()}}(jQuery);

    </script>
    <?php
  
}


if (!(is_admin())) {
wp_enqueue_script('jquery');
add_action('wp_head', 'jquery_parallax');
}

```
By adding the css selector 'parallax' in the Advanced tab creating a section, under Style setting a background image for it and under Attachment choosing Fixed I have the parallax effect I am looking for.

I'm not familiar enough with GitHub and can't easily find where the code for the background image of the section is to try and add it myself, so it'd be cool if you could implement this.

**Comment by ehausen** [comment]
Nowadays, to create a parallax effect javascript schouldn't be needed too much, other than for fallback.

**Pure CSS Parallax Websites**
http://keithclark.co.uk/articles/pure-css-parallax-websites/

**Paul Irish: Parallax Performance [CSSConfUS2014]**
https://www.youtube.com/watch?v=R6TXuXV1bbY

**Comment by ehausen** [comment]
I think this could be controlled in the same dropdown where you now can set background as default, fixed, scroll.

**Comment by joshallen73** [comment]
I would also love to see parallax. I use Astra wordpress theme with elementor..please put this feature request to the top. One thing beaver builder has that Im missing.

**Comment by marriott79** [comment]
Yes, please add a parallax option under the fixed/scroll drop-down. That would be great to have!

**Comment by simonclay** [comment]
A parallax option under the fixed/scroll drop-down would really be useful

**Comment by JulienBidart** [comment]
Hello

Thanks for your work.

I noticed that for Elementor the parralax effect some solutions are not working properly on Iphone like on desktop or Ipad.

On Iphone there is no zoom in for the image, it is contained in the section, so a repetition takes place.

Any help? I thought maybe someone could take the code SiteOrigin is using, cause their Parallax effect for their page builder works great on Iphone too. I am not an expert here thus.

Regards

Julien

**Comment by BetsyBurke** [comment]
I too would like to have an easy way to put parallax scrolling without having to write CSS for each image. Thanks.

**Comment by thewebstylist** [comment]
Pls PLS Elementor bring Mobile Parallax image background options to us ASAP!! That's been my fav site flavor for years

**Comment by Agam6130Haim6130** [comment]
Hi @arielk,

[Here](https://wordpress.org/support/topic/no-parallax-scroll/#post-9349102) you said that _"Parallax is one of the things we have at the top of our list of priorities. I think we will release this feature in the next versions."_

It was almost a year ago.
Please Ariel Achi-Hayakar, Parallax is an essential popular effect that you must add to sections&columns.
Untill last week elementor users still could use the "Granular Controls for Elementor", but it's no longer working with the last version of elementor.

Please add Parallax background option.

Thanks a lot!

**Comment by JoeDarko** [comment]
Any updates? Parallax effects are pretty essential theese days, it's a shame we don't have them in Elementor.

**Comment by hufer** [comment]
https://pafe.piotnet.com/#parallax-background-for-every-element Piotnet Addons For Elementor can do it for you

**Comment by RodrigoMingus** [comment]
Use a container element and add a background image to the container with a selected height. Then use the **### background-attachment: fixed** to make the particular parallax effect. the opposite background properties are wont to center and scale the image perfectly:

https://themexriver.com/downloads/category/html/

**Comment by dev-Asifemon** [comment]
> https://pafe.piotnet.com/#parallax-background-for-every-element Piotnet Addons For Elementor can do it for you

no it cant.

**Comment by mikeecho2** [comment]
Definitely not completed

Just to clarify from "Slider Widget fixed images #3342" 
Q: "Now the images scroll with the rest of the content but my client wanted them fixed like I can fix also the background-image of a section."
A: "Yea, well that is called parallax"

No it's not called parallax, its called fixed background, and option is still not available in Elementor, 5 years later
